---
title: "Analysis"
author: "Joaquín Rodrigo Ponce de León Conconi"
date: "4/30/2021"
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

First, the libraries needed are imported
```{r}
library(tidyverse) #To use certain functions like the pipe %>% 
library(stringr) #To process the string
```

The DNA coding strand sequence for Triakis scyllium IgNAR clone 3 is put as a string 
(reference [1] contains the sequence)
```{r}
ignar3_seq <- str_replace_all("CTGAGGTTTTCAGTGCTTGTGAAGAAACAATGCATATTTTCTGGGCTGCGCTCCTTCTAACGTGGTTATCAAATGCCTTTAGTGCACATGTCGATCAAACACCACGAGTGGCAACTAAAGAAACAGGCGAATCTCTGACA
ATCAATTGCGTCCTAAGGGGTGCTAGCTGTCTCTTGGACGCAACAAGCTGGTTTCGGCAAAATCCTGGTT
CAACAGGTTGGGAACGCATCACAATTGGCGGCCGATATGTTGACTCAGTCAACAAGGGAAGCAAGTCGTT
TTCTCTGCAAATCAAGGACCTAACAGTTGTAGACAGTGTCACATTTTACTGCACCGCTCAATACTACGTG
GGGCATGGCTGTTATGGGCTTGCAGTTGAAGATGGGGCTGGCACCGTGCTGACTGTGAATCCTGGACCGA
CTCCCCCAATCATCAATCTCTTCTCTGAAACTGATGAGTTGAGAGCAAAGGGGTTTGTGCAACTGATTTG
TTTGATCAGCGAGTACAAACCAGAAAGCATTAGAGTGAGCTGGGAGAAGAATGGAAACGCCCGACAGTCT
GGCTTCACCACTACGTCCCCATGTAAAACAGCAAAAGGTGAATTCCAATCCAGGAGCATCCTTACGTTGC
CGCTTCAGGAATGGAACAGCGGTTCCACGTACAGCTGCCAAGTGACCCATTCTGCTACCAACAGCAACAA
ACGAAAAGAAATTAGATCAACATCAGAAATCACAGTATTTCTCAGAGACCCCTCACTTGAAGAAATTTGG
ATAAGGAAAACTGTCACACTCATTTGTGAAGTGGTATCCACAGTTCCCAGTGTGGTTGGGATCTCTTGGA
CAGTGGATGGTAAGAAAAGGACCGAAGGTGTTCAAATTGAAGGACGTCAACAAGGGCAAAACCAGTATCT
GACCATCAGCCGCTTGACCAGCAGTGTGGAGGAGTGGGATCGTGGGGCTGAATACAACTGTTCAGCACAG
CAGAGCGAATCATCGACTCCAGTCTCAAAGCACACACAAAAGTTAAAAGTCAAGCCATCGAAGCCCAATC
TCCGACTCCTGCCTCCATCAGCAGAGGAGCTTCAAAGTTCCAGCGTCGCTACTCTCACTTGTTTGATAAG
AGGATTCTACCCGGACAAAATATCCATTTCCTGGGAGAAGGACGGAGCTGTTTTGAGCTCGAACATCACT
CGCTTCCCCACTGCTCTGGAACAGGACCAGACCTTCAGCACAAGCAGCCTCCTCATTTTACCGGCAGGGG
AATGGAAGACAGGAGCAAGATACACCTGTACCGCTTCACATCCAGCTTCAAAATTCACCGGAAAGAGGAC
AATCAACAGCCCGAAAGCTGATTGCTATGAGGAAGACATTTCCGTAAATATTTTGAACCCTTCATTTGAG
GAGATTTGGGTCCAAAAGACAGCGACAATTGTTTGTGAAATCCGTTACACTGTGTTAGAAAACGTCAGTG
TCTCCTGGCAAGTGGATGGGAGAATGAGAACGGAAGGAGTTGAGACCCAAACTCCTGAATGGAGTGGAAG
TAAAACAACGATCATGAGTAAACTGAAAGTCACAGCAGCAGAGTGGGACACTGGGGTGGAATATGTCTGC
CTCGCAGAAGGAAGTGAATTACCAACACCAAAGAAACGGTCCACTCGGAAAATAAAGGTTGGTGCAATGA
ATAGTCCTAAAGTGTACATTCTGCCACCATCAGTGGCCGAAATTGACTCAGAGAAGACGGCAACTTTGAT
GTGTCTAGCCACTGGCTTTTATCCTGCTGAAATTTATATCGCTTGGTTGGCCAATGACACGCTTTTAGAT
TCGGATTTCCCCAATCAACCAGTGAGCGAGAAAGGAAATGGTTCCAGCTTCATTGCCAGCAGATTGAGGC
TCACAGCGGCGGAGTGGAACACTGGCACAACTTATTCCTGTTTAGTGGGACACCCGTCCCTTGAAAGGAA
TTTAATCCGAAGTATAAATAAATCTTATGGTAAACCTACTTTAGTTAATGTTTCACTTGCCCTCGCTGAC
AGTTTTACATCTTGTGCATAACCCGCCACGACTGAATTGTTCTTTTCTGTAAAAGTGTCACCACCTTTCG
GTTCATTTTTTTTTGGTAATTGGAAACAGCAAATCTCACAATCACGCTTTAGAGCAGCAAATACTTGCTT
TGTCATTTTGCTAACACATAAAACT", "[\n]", "")
ignar3_seq

```


After seeing there are no \n, we need to know the length of the sequence (how many base pairs the gene has)
```{r}
ignar3_sp <- strsplit(ignar3_seq[1], "")
length(ignar3_sp[[1]])

```
As in GenBank, we can see that the gene indeed is still complete, since it is still 2195 bp long. So now we need to deduce the mRNA of this sequence.
```{r}
ignar3_mrn_sp <- c()
for (i in ignar3_sp[[1]]){
  if (i == "T"){
    ignar3_mrn_sp <- append(ignar3_mrn_sp, "U")
  }else{
    ignar3_mrn_sp <- append(ignar3_mrn_sp, i)
  }
}
length(ignar3_mrn_sp)
ignar3_mrn <- str_c(ignar3_mrn_sp, collapse = "")
ignar3_mrn
```

Now we can do the same for IgNAR clone 2
```{r}
ignar2_seq <- str_replace_all("GAAGTTTTCAGTGCTTGTGAAGAAACAATGCATATTTTCTGGGCTGCGCTCCTTCTAACGTGGTTATCAA
ATGCCTTTAGTGCACATGTCGATCAAACACCACGAGTGGCAACTAAAGGCACAGGCGAATCTCTGACAAT
CAATTGCGTCCTAAGGGGTGCTAGAAATGGCTTGTACGCAACAAGCTGGTTTCGGCAAAATCCTGGTTCA
ACAGGTTGGGAACGCATGACAATTGGCGGCCGATATATTGAATCAGTCACCAAGGGAAACAAGTCGTTTT
CTCTGCAAATCAAGGACCTAACAGTTGAAGACAGTGTCACCTTTTACTGCAAAGCTCAAGGGGATACTAC
GTGGGGGCTCGCTTCGGATGACTACTATGATGGGGCTGGCACCGTGCTGACTGTGAATCCTGGACCGACT
CCCCCAATCATCAATCTCTTCTCTGAAACTGATGAGTTGAGAGCAAAGGGGTTTGTGCAACTGATTTGTT
TGATCAGCGAGTACAAACCAGAAAGCATTAGAGTGAGCTGGGAGAAGAATGGAAACGCCCGACAGTCTGG
CTTCACCACTACGTCCCCATGTAAAACAGCAAAAGGAGAATTCCAATCCAGGAGCATCCTTACGTTGCCG
CTTCAGGAATGGAACAGCGGTTCCACGTACAGCTGCCAAGTGACCCATTCTGCTACCAACAGCAACAAAC
GAAAAGAAATTAGATCAACATCAGAAATCACAGTATTTCTCAGAGACCCCTCACTTGAAGAAATTTGGAT
AAAGAAAACTGTCACACTCATTTGTGAAGTGGTATCCACAGTTCCCAGTGTGGTTGGGATCTCTTGGACA
GTGGATGGTAAGAAAAGGACCGAAGGTGTTCAAATTGAAGGACGTCAACAAGGGCAAAACCAGTATCTGA
CCATCAGCCGCTTGACCAGCGGTGTGGAGGAGTGGGATCGTGGGGCTGAATACAACTGTTCAGCACAGCA
GAGCGAATCATCGACTCCAGTCTCAAAGCACACACAAAAGTTAAAAGTCAAGCCATCGAAGCCCAATCTC
CGACTCCTGCCTCCATCAGCAGAGGAGCTTCAAAGTTCCAGCGTCGCTACTCTCACTTGTTTGATAAGAG
GATTCTACCCGGACAAAATATCCATTTCCTGGGAGAAGGACGGAGCTGTTTTGAGCTCGAACATCACTCG
CTTCCCCACTGCTCTGGAACAGGACCAGACCTTCAGCACAAGCAGCCTCCTCATTTTACCGGCAGGGGAA
TGGAAGACAGGAGCAAGATACACCTGTACCGCTTCACATCCAGCTACAAAATTCACCGGAAAGAGGACAA
TCAACAGCCCGAAAGCTGATTGCTATGAGGAAGACATTTCCGTAAATATTTTGAACCCTTCATTTGAGGA
GATTTGGATCCAAAAGACAGCGACAATTGTTTGTGAAATCCGTTACACTGTGTTAGAAAACGTCAGTGTC
TCCTGGCAAGTGGATGGGAGAATGAGAACGGAAGGAGTTGAGACCCAAACTCCTGAATGGAGTGGAAGTA
AAACAACGATCATGAGTAAACTGAAAGCCACAGCAGCAGAGTGGGACACTGGGGTGGAATATGTCTGCCT
CGCAGAAGGAAGTGAATTACCAACACCAAAGAAACGGTCCACTCGGAAAATAAAGGTTGGTGCAATGAAT
AGTCCTAAAGTGTACATTCTGCCACCATCAGTGGCCGAAATTGACTCAGAGAAGACGGCAACTTTGATGT
GTCTAGCCACTGGCTTTTATCCTGCTGAAATTTATATCGCTTGGTTGGCCAATGACACGCTTTTAGATTC
GGATTTCCCCAATCAACCAGTGAGCGAGAAAGGAAATGGTTCCAGCTTCATTGCCAGCAGATTGAGGCTC
ACAGCGGCGGAGTGGAACACTGGCACAACTTATTCCTGTTTAGTGGGACACCCGTCCCTTGAAAGGAATT
TAATCCGAAGTATAAATAAATCTTATGGTAAACCTACTTTAGTTAATGTTTCACTTGCCCTCGCTGACAG
TTTTACATCTTGTGCATAACCCGCCACGACTGAATTGTTCTTTTCTGTAGAAGTGTCACCACCCTTTCGG
TTCATTTTTTTTGGGTAATTGGAAACAGCAAATCTCACAATCACGCTTTAGAGCAGCAAATACTTGCTTT
TGTCATTTTGCTAACACATAAAACTATACCATTT", "[\n]", "")
ignar2_sp <- strsplit(ignar2_seq[1], "")
ignar2_mrn_sp <- c()
for (i in ignar2_sp[[1]]){
  if (i == "T"){
    ignar2_mrn_sp <- append(ignar2_mrn_sp, "U")
  }else{
    ignar2_mrn_sp <- append(ignar2_mrn_sp, i)
  }
}
ignar2_mrn <- str_c(ignar2_mrn_sp, collapse = "")
length(ignar2_mrn_sp)
ignar2_mrn
```

REFERENCES

[1] Triakis scyllium IgNAR mRNA for immunoglobulin NAR, complete cds, clone: 3. (2010). NCBI Nucleotide. [online] Available at: https://www.ncbi.nlm.nih.gov/nuccore/AB557742.1?report=fasta [Accessed 1 May 2021].

[2] Triakis scyllium IgNAR mRNA for immunoglobulin NAR, complete cds, clone: 2. (2010). NCBI Nucleotide. [online] Available at: https://www.ncbi.nlm.nih.gov/nuccore/AB557741.1?report=fasta [Accessed 1 May 2021].




